test/src/k-provider/ovp/provider-parser.spec.js

```diff
@@ -35,6 +38,38 @@ describe('provider parser', function() {
       (adaptiveSource === null).should.be.true;
     });
   });
+  describe('_parseProgressiveSource', () => {
+    it('should return a valid progressive sources when getting separate http/s', () => {
+      const context = new playbackContext({});
+      context.flavorAssets = kalturaDashSourceFlavorAssets;
+      const progressiveSource = OVPProviderParser._getParsedSources(
+        [kalturaProgressiveSourceNotSecured, kalturaProgressiveSourceSecured],
+        'myKS',
+        1234,
+        1234,
+        {
+          id: '1_938734'
+        },
+        context
+      );
+      progressiveSource.should.exist;
+    });
+    it('should return a valid progressive source for a valid input', () => {
+      const context = new playbackContext({});
+      context.flavorAssets = kalturaDashSourceFlavorAssets;
+      const progressiveSource = OVPProviderParser._getParsedSources(
+        [kalturaProgressive],
+        'myKS',
+        1234,
+        1234,
+        {
+          id: '1_938734'
+        },
+        context
+      );
+      progressiveSource.should.exist;
```

Comment:
same here - don't we want to check the the selected source is the one you would expect? should equal protocol or deliveryProfileId?